pkg/server/commands/list_objects.go

```diff
@@ -303,13 +303,19 @@ func (q *ListObjectsQuery) evaluate(
 	return nil
 }
 
-func trySendObject(object string, objectsFound *uint32, maxResults uint32, resultsChan chan<- ListObjectsResult) {
-	if objectsFound != nil && atomic.AddUint32(objectsFound, 1) > maxResults {
-		return
+func trySendObject(object string, objectsFound *atomic.Uint32, maxResults uint32, resultsChan chan<- ListObjectsResult) {
+	if !getAllResults(maxResults) {
+		if objectsFound.Add(1) > maxResults {
+			return
+		}
 	}
 	resultsChan <- ListObjectsResult{ObjectID: object}
 }
 
+func getAllResults(maxResults uint32) bool {
+	return maxResults == 0
```

Comment:
imho this function doesn't add any more clarity than just doing the check where needed

Reply:
Agree